src/driver/main.cpp

```diff
@@ -569,11 +572,42 @@ struct verify : command<verify>
         auto t = c.ct.get_target();
         auto m = c.parameters.generate(p, t, true, c.l.batch);
 
+        // TODO remove this and make the driver able to figure out datatype most used in the model
+        //  then set the tolerances appropriately. Need to check here because c.to_fp16 only set
+        //  after argument_parser.parse() is run. This code is complicated because there's not a
+        //  good way to change the default tolerances after reading `--fp16` but before reading
+        //  `--rms-tol`, `--atol`, and `--rtol`.
+        migraphx::verify::tolerance tols{};
+        if(c.to_fp16)
+        {
+            tols = migraphx::verify::tolerance{8e-2, 4e-2, 4e-2};
```

Comment:
The tolerance should be set int `verify_program`, `verify_instructions`, and `verify_reduced_program` based on whether the model contains fp16 or not, as this wont set the tolerances correctly for prequantized models. We could probably right a function like `get_tolerances`:

```cpp
verify::tolerance get_tolerances(const program& p, std::optional<double> rms_tol, std::optional<double> atol, std::optional<double> rtol)
{
    verify::tolerance result{};
    bool has_fp16 = any_of(p.get_modules(), [](module_ref m) {
        return any_of(m, [](auto&& ins) {
            return ins.get_shape().type() == shape::half_type;
        });
    });
    if(has_fp16)
    {
        result.rms_tol = 8e-2;
        result.atol = 4e-2;
        result.rtol = 4e-2;
    }
    if(rms_tol)
        result.rms_tol = *rms_tol;
    if(atol)
        result.atol = *atol;
    if(rtol)
        result.rtol = *rtol;
    return result;
}
```

That can be called from these functions to set the tolerance.